### PR TITLE
Clarify description dimension_types + small corrections

### PR DESCRIPTION
**Pull request by JPBergsma**
I tried to clarify the definition of the `dimension_types` field as described in issue #390.
I made several small corrections and replaced pertains with belongs, as belongs is a more common word in English.

### CODE REVIEW

File: optimade.rst [hunk 1]
```diff
@@ -1564,7 +1564,7 @@ OPTIONAL features:
 Examples:
 
 - :filter:`chemical_formula_anonymous CONTAINS "C2" AND chemical_formula_anonymous STARTS WITH "A2"`
-- :filter:`chemical_formula_anonymous STARTS "B2" AND chemical_formula_anonymous ENDS WITH "D2"`
+- :filter:`chemical_formula_anonymous STARTS WITH "A2" AND chemical_formula_anonymous ENDS WITH "D2"`
```
Comment (merkys) [inline optimade.rst:1567]: Great catch!

**Review by merkys — COMMENTED** [review]
This PR contains a bundle of not-so-much related changes, some of them are straightforward, some are a bit more controversial thus may slow down PR's acceptance. I cannot comment on "pertains" -> "belongs" change as non-natural English speaker. Defining `dimension_types` as "periodicity of boundaries" sounds good to me as I know the original intent, but it would be great to hear other opinions as well.

File: optimade.rst (hunk 1 above)
Comment (ml-evs) [reply optimade.rst:1567]: I think the "WITH" is optional anyway: https://github.com/Materials-Consortia/OPTiMaDe/blob/develop/optimade.rst#623substring-comparisons

> identifier STARTS WITH x: Is true if the property starts with the substring value x. The WITH keyword MAY be omitted.

Comment (merkys) [reply optimade.rst:1567]: Oh, right! Then there is no need to change that. We would probably want our specification to showcase also the alternative syntax constructs.

Comment (JPBergsma) [reply optimade.rst]: Ok, I removed "WITH".
I also changed "D2" to "D1" as a chemical_formula_anonymous can never start with "A2" and end with "D2".

File: optimade.rst [hunk 2]
```diff
@@ -2004,10 +2004,10 @@ dimension\_types
 
 - **Examples**:
 
-  - For a molecule: :val:`[0, 0, 0]`
-  - For a wire along the direction specified by the third lattice vector: :val:`[0, 0, 1]`
-  - For a 2D surface/slab, periodic on the plane defined by the first and third lattice vectors: :val:`[1, 0, 1]`
-  - For a bulk 3D system: :val:`[1, 1, 1]`
+  - A non periodic unit cell, for example, for a single molecule : :val:`[0, 0, 0]`
```
Comment (ml-evs) [inline optimade.rst:2007]:
```suggestion
  - A nonperiodic structure, for example, for a single molecule : :val:`[0, 0, 0]`
```

File: optimade.rst [hunk 3]
```diff
@@ -2004,10 +2004,10 @@ dimension\_types
...
+  - A unit cell that is periodic in the direction of the third lattice vector, for example for a carbon nanotube: :val:`[0, 0, 1]`
+  - For a 2D surface/slab, who's unit cell is periodic in the direction of the first and third lattice vectors: :val:`[1, 0, 1]`
```
Comment (ml-evs) [inline optimade.rst:2009]:
```suggestion
  - For a 2D surface/slab, with a unit cell that is periodic in the direction of the first and third lattice vectors: :val:`[1, 0, 1]`
```

**Review by ml-evs — APPROVED** [review]

**Review by merkys — APPROVED** [review]

**Pull request merged**